src/Umbraco.Core/Models/IRelation.cs

```diff
@@ -4,37 +4,81 @@
 
 namespace Umbraco.Core.Models
 {
-    public interface IRelation : IEntity, IRememberBeingDirty
+    public interface IRelation :  IRememberBeingDirty, IRelationReadOnly
     {
+
+        /// <summary>
+        /// Gets or sets the integer identifier of the entity.
+        /// </summary>
+        new int Id { get; set; }
+
         /// <summary>
         /// Gets or sets the Parent Id of the Relation (Source)
         /// </summary>
         [DataMember]
-        int ParentId { get; set; }
+        new int ParentId { get; set; }
 
         [DataMember]
-        Guid ParentObjectType { get; set; }
+        new Guid ParentObjectType { get; set; }
 
         /// <summary>
         /// Gets or sets the Child Id of the Relation (Destination)
         /// </summary>
         [DataMember]
-        int ChildId { get; set; }
+        new int ChildId { get; set; }
 
         [DataMember]
-        Guid ChildObjectType { get; set; }
+        new Guid ChildObjectType { get; set; }
 
         /// <summary>
         /// Gets or sets the <see cref="RelationType"/> for the Relation
         /// </summary>
         [DataMember]
-        IRelationType RelationType { get; set; }
+        new IRelationType RelationType { get; set; }
 
         /// <summary>
         /// Gets or sets a comment for the Relation
         /// </summary>
         [DataMember]
-        string Comment { get; set; }
+        new string Comment { get; set; }
+
+        /// <summary>
+        /// Gets the Id of the <see cref="RelationType"/> that this Relation is based on.
+        /// </summary>
+        [IgnoreDataMember]
+        new int RelationTypeId { get; }
+    }
+
+    public interface IRelationReadOnly : IEntity
```

Comment:
What I was proposing was: 

> a new ReadOnlyRelation (or something) that is a simplified/slim version of what is needed to create/update a relation which doesn't need to implement all of the IEntity/IRememberBeingDirty, etc... since it doesn't need anything and all of it's properties are readonly

This shouldn't inherit from IEntity since it doesn't need to be and it's underlying properties are mutable (and also contain Id). It should just contain the properties needed to persist a relation without inheriting from anything. Then IRelation  should not inherit from this, the `new` hiding properties is pretty ugly IMO.

Reply:
Please re-review with latest changes